custom_components/heatmiserneo/climate.py

```diff
@@ -164,76 +137,55 @@ def hvac_modes(self):
         """Return the list of available operation modes."""
         return self._hvac_modes
 
-    # @property
-    # def preset_mode(self):
-    #     """Return preset mode."""
-    #     return self._preset
-
-    # @property
-    # def preset_modes(self):
-    #     """Return preset modes."""
-    #     return self._preset_modes
-
-    def set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs):
         """ Set new target temperature. """
-        response = self._hub.json_request({"SET_TEMP": [int(kwargs.get(ATTR_TEMPERATURE)), self._name]})
+        response = await self._hub.set_temp(
+            int(kwargs.get(ATTR_TEMPERATURE)), [self._neostat]
```

Comment:
We should probably drop `int` here. `int` will break support for setting temperature with 0.5 degree accuracy.

I should probably add some function to neohubapi that returns `target_temperature_step`, so we can round to appropriate precision... But I think simply dropping `int` would work for now.